aspnetcore/grpc/basics.md

```diff
@@ -0,0 +1,61 @@
+---
+title: gRPC services with C#
+author: juntaoluo
+description: Learn the basic concepts when writing gRPC serices with C#.
+monikerRange: '>= aspnetcore-3.0'
+ms.author: johluo
+ms.date: 03/08/2019
+uid: grpc/basics
+---
+# gRPC services with C#
+
+This document outlines the basic concepts needed to write gRPC apps in C#. The topics covered here apply to both C Core based and ASP.NET Core based apps.
+
+## `.proto` file
+
+gRPC uses a design-first approach to API development. Protocol buffers (protobuf) are used as the Interface Design Language (IDL) by default. As a result, `.proto` file serve as the definition of the gRPC service and messages sent between clients and servers. For more information on the syntax of protobuf files, visit the [official documentation (protobuf)](https://developers.google.com/protocol-buffers/docs/proto3)).
+
+For example, the `greet.proto` file defines a `Greeter` service which defines a `SayHello` call that sends a `HelloRequest` message and receives a `HelloResponse` message:
+
+[!code-proto[](~/tutorials/grpc/grpc-start/samples/GrpcStart/Protos/greet.proto)]
+
+## Add a `.proto` file to your C# app
+
+The `.proto` file is included in your project by adding it to the `<ProtoBuf>` item group:
```

Comment:
`ProtoBuf` is used here vs `Protobuf` in the code. Which is the correct one?

Reply:
Yea I had the same question in https://github.com/grpc/grpc-dotnet/pull/48#discussion_r251773522. This item group is going to be renamed: https://github.com/grpc/grpc/issues/17884. I guess we can preempt the change and use all lowercase. MSBuild ignores case anyway.